### ISSUE TITLE
Non-transient error: <nil>

### ISSUE THREAD

**Issue by black-snow** [argoproj/argo-workflows#13881]
### Pre-requisites

- [X] I have double-checked my configuration
- [X] I have tested with the `:latest` image tag (i.e. `quay.io/argoproj/workflow-controller:latest`) and can confirm the issue still exists on `:latest`. If not, I have explained why, **in detail**, in my description below.
- [X] I have searched existing issues and could not find a match for this bug
- [X] I'd like to contribute the fix myself (see [contributing guide](https://github.com/argoproj/argo-workflows/blob/main/docs/CONTRIBUTING.md))

### What happened? What did you expect to happen?

I encounter a lot of these:

```
Non-transient error: <nil>
```

and sadly there's not much context I can give. Actually, I can give close to `nil` context. ;)

We run workflows with up to 250 nodes and thousands of concurrent pods. The above log is apparenty just a `warning` but the `nil` there seems worrisome. Should I worry?

Sadly, I cannot provide proper steps to reproduce. This ain't very helpful but perhaps someone else is seeing the same or has an immediate idea.

### Version(s)

v3.5.12

### Paste a minimal workflow that reproduces the issue. We must be able to run the workflow; don't enter a workflows that uses private images.

```YAML
N/A
```

### Logs from the workflow controller

```text
time="2024-11-08T10:02:19.639Z" level=warning msg="Non-transient error: <nil>"
```

### Logs from in your workflow's wait container

```text
N/A
```

**Comment by tooptoop4** [comment]
@black-snow provide the controller logs surrounding that line

**Comment by isubasinghe** [comment]
This is not really actually an issue,  there is a bug where we call `IsTransientError` on an nil error.

**Comment by black-snow** [comment]
Thanks for looking into this. Do you still need the surrounding logs?
Can we get rid of them if they are false positives?

**Comment by sarabala1979** [comment]
@black-snow Do you like to contribute? It is a good first issue

**Comment by tooptoop4** [comment]
@black-snow the surrounding logs can help to know which line to change

**Comment by black-snow** [comment]
@sarabala1979 absolutely. Sounds like @isubasinghe already has a notion of where to look.

**Comment by MasonM** [comment]
@black-snow No need for the logs. I can reproduce this locally and entered https://github.com/argoproj/argo-workflows/pull/13917 with a fix.
